lib/registries/addon/edit-revision/revision-manager.ts

```diff
@@ -236,6 +242,21 @@ export default class RevisionManager {
         }
     }
 
+    @restartableTask
+    @waitFor
+    async saveWithToast() {
+        try {
+            taskFor(this.onJustificationInput).cancelAll();
+            taskFor(this.onPageInput).cancelAll();
+            await taskFor(this.updateRevisionAndSave).perform();
+            await taskFor(this.saveAllVisitedPages).perform();
```

Comment:
Are there any problems if the first one completes but the second one does not? I'm not super-familiar with this workflow.

Reply:
If the second task fails, it should just catch and show the error toast. Is there something I'm overlooking?

Reply:
It's probably fine. Like I say, I don't know about this particular workflow, I just saw you had two complex things that had a potential to fail in one try-catch block, and I wasn't sure if that would cause a weird state that you'd want to handle differently somehow. If not, no worries.